packages/dx-grid-core/src/plugins/grouping-state/computeds.js

```diff
@@ -72,3 +74,37 @@ export const nextExpandedGroups = (prevExpandedGroups, groupKey) => {
 
   return expandedGroups;
 };
+
+const ungroupedColumnIndex = (prevGrouping, nextGrouping) => {
+  if (prevGrouping.length > nextGrouping.length) {
+    for (let i = 0; i < prevGrouping.length; i += 1) {
```

Comment:
A complexity of this algorithm is too big. Feels like this function can be written with O(n) complexity.